gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/TimingEventUtils.java

```diff
@@ -47,6 +47,13 @@ static Map<String, String> getFlowMetadata(Config flowConfig) {
     return metadata;
   }
 
+  /**
+   * Retrieves a flowExecutionId from flowMetadata map and returns dummy value if one is not set
+   */
+  public static String getFlowExecutionIdFromFlowMetadata(Map<String, String> flowMetadata) {
+    return flowMetadata.getOrDefault(TimingEvent.FlowEventConstants.FLOW_EXECUTION_ID_FIELD, "<<no flowExecutionId>>");
+  }
```

Comment:
I really like the abstraction!  as far as where this should live... don't we already have another utils class with static methods for extracting the flow group and flow name?

Reply:
This class has methods relating to this map of `FlowMetadata` specifically. It uses a different key than the normal flow group/name/execution id keys from `ConfigurationKeys`. I'm keeping related functionality in one place here

Reply:
makes sense.  I now see that the `DagManagerUtils` is unfortunately only applicable to that specific context... not also here... :(